reference/data-consistency.md

```diff
@@ -0,0 +1,78 @@
+---
+description: Description of Turso data consistency guarantees.
+keywords:
+  - turso
+  - technical
+  - reference
+  - consistency
+---
+
+# Data consistency
+
+Turso is built on top of [libSQL], which is a fork of [SQLite]. SQLite, as an
+embedded database, offers a strictly serializable consistency model. However,
+libSQL, as a network-accessible and replicated database, can not offer such
+a strong guarantee.
+
+## Connections
+
+Clients may establish a connection to a sqld instance, and each instance may
+handle a number of client connections concurrently. For the purpose of this
+documentation, a connection to a sqld instance is referred to as a *process*,
+which issues an ordered sequence of queries with results.
```

Comment:
What does "connection" mean; a Hrana connection? We should also clarify the semantics for HTTP requests, too.

Reply:
Marin confirms that with HTTP requests, the guarantees are much weaker; we can't even guarantee read-your-writes on replicas. @MarinPostma

Reply:
After discussing this with Marin, we realized that "connection" here corresponds to a single call to `Client.execute()`, `Client.batch()` or `Client.transaction()` in the client SDKs.

In particular, two subsequent calls to `Client.execute()` are different "connections", so the read-your-write guarantee no longer applies. You can do `await client.execute("INSERT ..."); await client.execute("SELECT ...");` and the SELECT is not guaranteed to observe the effect of the INSERT.

Reply:
Just adding a bit of clarification here: the read-your-write guarantee applies to a single database connection. Since the client has HTTP semantics in mind, and each call is considered independent (it can hit another replica or whatnot), each execute is performed on a different connection. A connection is what identifies a process to the database.

Reply:
Yes; each call to `execute()`, `batch()` and `transaction()` creates a new "connection".